rpc/src/v1/tests/helpers/signature.rs

```diff
@@ -0,0 +1,57 @@
+use std::sync::Arc;
+use ethcore::account_provider::AccountProvider;
+use v1::helpers::verify_signature;
+use v1::helpers::dispatch::eth_data_hash;
+use hash::keccak;
+
+pub fn add_chain_replay_protection(v: u64, chain_id: Option<u64>) -> u64 {
+	v + if let Some(n) = chain_id { 35 + n * 2 } else { 0 }
+}
+
+fn setup(is_prefixed: bool, signining_chain_id: Option<u64>, rpc_chain_id: Option<u64>, is_valid_for_current_chain: bool) {
+	let accounts = Arc::new(AccountProvider::transient_provider());
+
+	let address = accounts.new_account(&"password123".into()).unwrap();
+	let data = vec![5u8];
+	let  hash = if is_prefixed { eth_data_hash(data.clone()) } else { keccak(data.clone()) };
+	let sig = accounts.sign(address, Some("password123".into()), hash).unwrap();
+	let (v, r, s) = (sig.v(), sig.r(), sig.s());
+	let v = add_chain_replay_protection(v as u64, signining_chain_id);
+	let account = verify_signature(is_prefixed, data.into(), v, r.into(), s.into(), rpc_chain_id).unwrap();
+	assert_eq!(account.inner.address, address);
+	assert_eq!(account.inner.is_valid_for_current_chain, is_valid_for_current_chain)
+}
+
+#[test]
+fn test_verify_signature_prefixed_mainnet() {
+	setup(true, Some(1), Some(1), true)
```

Comment:
Test in a current form is pretty unreadable - I think it would be way better to avoid passing so many bools and numbers without some additional description.
I think the `setup` should only be producing the signature, the assertion and call to `verify_signature` should be inlined and duplicated in every test, like this:

```rust
#[test]
fn test_verify_signature_not_prefixed_mainnet() {
  let should_prefix = true;
  let data = vec![5u8];
  let chain_id = Some(1);
  let (address, v, r, s) = sign(should_prefix, &data, chain_id);

  let account = verify_signature(should_prefix, &data, v, r, s, chain_id).unwrap();

  assert_eq!(account.inner.address, address);
  assert_eq!(account.inner.is_valid_for_current_chain, true);
}
```

IMHO that would really help readability/understandability of the test. If you want to keep the de-duplication, then consider using this:

```rust
#[test]
fn test_verify_signature_not_prefixed_mainnet() {
  run_test(TestCase {
    should_prefix: true,
    signing_chain_id: Some(1),
    rpc_chain_id: Some(1),
    is_valid_for_current_chain: true,
  });
}
```